src/pages/settings/ProfilePage/LoginField.js

```diff
@@ -0,0 +1,101 @@
+    // Label to display on login form
+    label: PropTypes.string.isRequired,
+
+    // Type associated with the login
+    type: PropTypes.oneOf([CONST.LOGIN_TYPE.EMAIL, CONST.LOGIN_TYPE.PHONE]).isRequired,
+
+    // Login associated with the user
+    login: PropTypes.shape({
+        partnerUserID: PropTypes.string,
+        validatedDate: PropTypes.string,
+    }).isRequired,
+};
+const LoginField = ({
+    label,
+    login,
+    type,
+}) => {
+    let note;
+    if (type === CONST.LOGIN_TYPE.PHONE) {
+        // No phone number
+        if (!login.partnerUserID) {
+            note = 'Add your phone number to settle up via Venmo.';
+
+            // Has unvalidated phone number
+        } else if (!login.validatedDate) {
+            note = 'The number has not yet been validated. Click the button to resend the validation link via text.';
+
+            // Has verified phone number
+        } else {
+            note = 'Use your phone number to settle up via Venmo.';
+        }
+
+        // Has unvalidated email
+    } else if (login.partnerUserID && !login.validatedDate) {
+        note = 'The email has not yet been validated. Click the button to resend the validation link via text.';
+    }
+
+    return (
+        <View style={styles.mb6}>
+            <Text style={styles.formLabel}>{label}</Text>
+            {!login.partnerUserID ? (
+                <Pressable
+                    style={[styles.createMenuItem, styles.ph0]}
+                    onPress={() => Navigation.navigate(type === CONST.LOGIN_TYPE.PHONE
+                        ? ROUTES.SETTINGS_ADD_PHONE
+                        : ROUTES.SETTINGS_ADD_EMAIL)}
+                >
+                    <View style={styles.flexRow}>
+                        <View style={styles.createMenuIcon}>
+                            <Icon src={Plus} />
+                        </View>
+                        <View style={styles.justifyContentCenter}>
+                            <Text style={[styles.createMenuText, styles.ml3]}>
+                                {`Add ${label}`}
+                            </Text>
+                        </View>
+                    </View>
+                </Pressable>
+            ) : (
+                <View style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter]}>
+                    <Text style={[styles.textP]} numberOfLines={1}>
+                        {login.partnerUserID}
+                    </Text>
+                    {!login.validatedDate && (
+                        <Pressable
+                            style={styles.button}
+                            onPress={() => resendValidateCode(login.partnerUserID)}
+                        >
+                            <Text style={styles.createMenuText}>
+                                Resend
+                            </Text>
+                        </Pressable>
+                    )}
+                </View>
+            )}
+            {note && (
```

Comment:
Hmm, quick question: prior to being validated, shouldn't this field continue to be editable? There's a chance the user mistypes their number or email, so I think they should still be able to amend the field here.

Reply:
This is something we discussed a while back in the doc and I believe we settled on it being a v2 for allowing users to remove secondary logins from this view. 

Editing it before resending the validation link would get into a pretty tricky flow since they'd effectively be starting over not resending a validation link.

Reply:
right now in e.com we don't allow any sort of modification to secondary logins beyond adding them, removing them, or making them primary logins. If you mistype, you just remove the login and start again.

Reply:
So there's a way to remove this then?

Reply:
In E.com yes, but for E.cash it's considered a v2 issue

Reply:
Okay, thank you for that context. In that case, can the padding (8px) be based on the button and then the phone number is center-aligned with the button? Hard to describe, so provided a visual below:
![image](https://user-images.githubusercontent.com/43361905/113224583-a3498a00-9240-11eb-8aa9-2401d1ba49d3.png)

Reply:
@michelle-thompson it gets a little tricky because the button isn't always there but i'll play around with it